util/health/disk_default.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !netbsd
+// +build !openbsd
+// +build !windows
+
+package health
+
+import (
+	"errors"
+	"syscall"
+)
+
+func (d *dsReporter) Report() error {
+	fs := syscall.Statfs_t{}
+	err := syscall.Statfs(d.path, &fs)
+	if err != nil {
+		return err
+	}
+	if fs.Bavail == 0 {
```

Comment:
does that work with quotas ?

Reply:
You mean [Linux disk quotas](https://www.centos.org/docs/5/html/Deployment_Guide-en-US/ch-disk-quotas.html)? I checked quickly with an ext4 fs and apparently quotas aren't taken into account (but the same is true for `df`).